src/Program.cs

```diff
@@ -1,245 +1,263 @@
+    // This is discussed in detail here: https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal
 
     class Program
     {
-        // You can obtain this information from the Azure management portal. The instructions in the link above 
-        // include details for this as well.
-        private const string TenantId = "<your tenant id>";
-        private const string ClientId = "<your client id>";
-        private const string SubscriptionId = "<your subscription id>";
-
-        // This is the return URL you configure during AD client application setup. For this type of apps (non-web apps)
-        // you can set this to something like http://localhost/testapp. The important thing is that the URL here and the
-        // URL in AD configuration match.
-        private static readonly Uri RedirectUrl = new Uri("<your redirect url");
+        public static void Main(string[] args)
+        {
+            IConfigurationBuilder builder = new ConfigurationBuilder().AddJsonFile("appsettings.json");
+            IConfigurationRoot configuration = builder.Build();
 
-        private static readonly Random _random = new Random();
+            // Updated this information in the appsettings.json in this project.
+            // See the instructions at the link above for the service principal details and the subscription id.
+            var tenantId = configuration["TenantId"];
+            var clientId = configuration["ClientId"];
+            var clientSecret = configuration["ClientSecret"];
+            var subscriptionId = configuration["SubscriptionId"];
 
-        private static string _authorizationToken = null;
+            if (new List<string> { tenantId, clientId, clientSecret, subscriptionId }.Any(i => String.IsNullOrEmpty(i)))
+            {
+                Console.WriteLine("Please provide values for tenantId, clientId, secret and subscriptionId.");
+            }
+            else
+            {
+                RunSample(tenantId, clientId, clientSecret, subscriptionId).Wait();
+            }
+        }
 
-        static void Main(string[] args)
+        public static async Task RunSample(string tenantId, string clientId, string clientSecret, string subscriptionId)
         {
-            HttpResponseMessage response;
+            // Build the service credentials using the service principal.
+            var creds = await ApplicationTokenProvider.LoginSilentAsync(tenantId, clientId, clientSecret);
+
+            // Create the various management clients using the credentials instantiated above.
+            var subscriptionClient = new SubscriptionClient(creds);
+            var resourceClient = new ResourceManagementClient(creds);
+            resourceClient.SubscriptionId = subscriptionId;
+            var searchClient = new SearchManagementClient(creds);
+            searchClient.SubscriptionId = subscriptionId;
+
+            var _random = new Random();
```

Comment:
nit: Normally it's only fields that are prefixed with underscore, not local variables.

Reply:
fixed